### PR TITLE
iamb: 0.0.9 -> 0.0.9-unstable-2024-08-15

### PR DESCRIPTION
**Pull request by aktaboot**
## Description of changes

Fixes iamb failing to build

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Nanotwerp — APPROVED** [review]
LGTM! Tested with `nix repl`:

```
nix-repl> :b inputs.test.legacyPackages.x86_64-linux.iamb

This derivation produced the following outputs:
  out -> /nix/store/23c6lczn83qhcggi58qccb62c5aq25kn-iamb-0.0.9-unstable-2024-08-15
```

**Comment by a-kenji** [comment]
There has been a release now:  #336278

File: pkgs/applications/networking/instant-messengers/iamb/default.nix [hunk 1]
```diff
@@ -8,16 +8,16 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "iamb";
-  version = "0.0.9";
+  version = "0.0.9-unstable-2024-08-15";
```
Comment (Nanotwerp) [inline pkgs/applications/networking/instant-messengers/iamb/default.nix:11]:
```suggestion
  version = "0.0.10";
```

File: pkgs/applications/networking/instant-messengers/iamb/default.nix [hunk 2]
```diff
@@ -8,16 +8,16 @@
...
 
   src = fetchFromGitHub {
     owner = "ulyssa";
     repo = "iamb";
-    rev = "v${version}";
-    hash = "sha256-UYc7iphpzqZPwhOn/ia7XvnnlIUvM7nSFBz67ZkXmNs=";
+    rev = "54cb7991be526e2b8cd45e0a5b7cdb4a6d2e310b";
```
Comment (Nanotwerp) [inline pkgs/applications/networking/instant-messengers/iamb/default.nix:16]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/networking/instant-messengers/iamb/default.nix [hunk 3]
```diff
@@ -8,16 +8,16 @@
...
+    hash = "sha256-CepMojqLh0VUnHkagMKGHnEJfA8/EIoGBVN2rRNyFko=";
```
Comment (Nanotwerp) [inline pkgs/applications/networking/instant-messengers/iamb/default.nix:17]:
```suggestion
    hash = "sha256-cjBSWUBgfwdLnpneJ5XW2TdOFkNc+Rc/wyUp9arZzwg=";
```

File: pkgs/applications/networking/instant-messengers/iamb/default.nix [hunk 4]
```diff
@@ -8,16 +8,16 @@
...
   };
 
-  cargoHash = "sha256-982FdK6ej3Bbg4R9e43VSwlni837ZK4rkMkoeYMyW8E=";
+  cargoHash = "sha256-MBi6R/hyIvPvTet4IohgFWgRZQQTnxuS2JV3kfqxoyA=";
```
Comment (Nanotwerp) [inline pkgs/applications/networking/instant-messengers/iamb/default.nix:20]:
```suggestion
  cargoHash = "sha256-a5y8nNFixOxJPNDOzvFFRqVrY2jsirCud2ZJJ8OvRhQ=";
```

**Review by Nanotwerp — CHANGES_REQUESTED** [review]
Version [`0.0.10`](https://github.com/ulyssa/iamb/releases/tag/v0.0.10) of `iamb` has just been released, so let's bump it to that.

**Comment by meain** [comment]
https://github.com/NixOS/nixpkgs/pull/336278 might be a better option.

**Comment by Nanotwerp** [comment]
#336278 has been merged!

**Pull request closed without merging**